libminifi/test/unit/IdTests.cpp

```diff
@@ -199,3 +288,77 @@ TEST_CASE("Test Hex Device Segment 18 bits", "[id]") {
   REQUIRE(true == LogTestController::getInstance().contains("Using minifi uid prefix: 9af8"));
   LogTestController::getInstance().reset();
 }
+
+
+TEST_CASE("Collision", "[collision]") {
+  TestController test_controller;
+
+  LogTestController::getInstance().setDebug<utils::IdGenerator>();
+  std::shared_ptr<minifi::Properties> id_props = std::make_shared<minifi::Properties>();
+  SECTION("random") {
+    id_props->set("uid.implementation", "random");
+  }
+  SECTION("time") {
+    id_props->set("uid.implementation", "time");
+  }
+  SECTION("uuid_default") {
+    id_props->set("uid.implementation", "uuid_default");
+  }
+
+  std::shared_ptr<utils::IdGenerator> generator = utils::IdGenerator::getIdGenerator();
+  generator->initialize(id_props);
+
+  std::vector<utils::Identifier> uuids(16 * 1024U);
+  std::vector<std::thread> threads;
+  for (size_t i = 0U; i < 16U; i++) {
+    threads.emplace_back([&, i](){
```

Comment:
I think & capture includes "i" as well.

Reply:
It does, but I explicitly want to capture it by value to make all threads have a unique one. I will rewrite this to explicitly capture everything it needs to make it more readable.